include/net/ethernet.h

```diff
@@ -33,6 +33,10 @@
 
 #define ETHER_ADDR_LEN  6
 
+#define ETHERTYPE_ARP   0x0806 /* Address resolution protocol */
+#define ETHERTYPE_IP    0x0800 /* IP */
+#define ETHERTYPE_IPV6  0x86dd /* IP protocol version 6 */
```

Comment:
why defines from `include/nuttx/net/ethernet.h` can't be used?

Reply:
glibc and musl both define ETHERTYPE_*, so many applications use this definition, and if we compile directly in NuttX's environment, we will get a compile error.

Reply:
Maybe the defines in `include/nuttx/net/ethernet.h` should be moved here, to be similar to glibc and musl in this regard? (FreeBSD has it in sys/net/ethernet.h -- I am not sure if that's the include path programs use, or if they use net/ethernet.h without the sys.)

Reply:
Another idea: With Kconfig, developers could select: NuttX POSIX-only, Linux-style, and/or BSD-style. These would not be mutually exclusive; that is, you could select more than one. Those selection(s) would enable/disable things like glibc-style declarations. Two disadvantages of this idea: (1) more Kconfig/ifdef in the codebase, (2) how do we handle cases where Linux-style and BSD-style conflict with each other?

Reply:
If no conflict happens, what's benefit to add option?